src/engine/init.c

```diff
@@ -46,7 +44,7 @@ static char		modules[MAX_MODULE_OPTIONS + 1];
 static unsigned int	nr_threads;
 
 /** DAOS system name (corresponds to crt group ID) */
-static char	       *daos_sysname = DAOS_DEFAULT_SYS_NAME;
+char                    daos_sysname[DAOS_SYS_NAME_MAX + 1] = DAOS_DEFAULT_SYS_NAME;
```

Comment:
[Question] Why is this change to array necessary?

Reply:
to allow for a sys name that might be larger than DAOS_DEFAULT_SYS_NAME and could be up to DAOS_SYS_NAME_MAX

Reply:
`daos_sysname` was a pointer, it could be assigned the address of a string longer than `DAOS_DEFAULT_SYS_NAME`, is that not so?

Reply:
yea, i'm not sure then why i made that change :-)
I'll revert it back on my follow on PR.